### PR TITLE
Clean up commands

### PR DESCRIPTION
**Pull request by Electroid**
Also fixes a bug where team sizes were not enforced before the match starts (oops!)

### CODE REVIEW

**Comment by KingOfSquares** [comment]
Closes #417

File: core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java [hunk 1]
```diff
@@ -48,6 +46,13 @@
 
 public class ChatDispatcher implements Listener {
 
+  private static ChatDispatcher INSTANCE = new ChatDispatcher();
```
Comment (KingOfSquares) [inline core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java:49]: `private static final`

File: core/src/main/java/tc/oc/pgm/PGMPlugin.java [hunk 2]
```diff
@@ -351,133 +312,29 @@ public VanishManager getVanishManager() {
...
-
-      node.registerCommands(vanishManager);
-      registerEvents((Listener) vanishManager);
-
-      node.registerCommands(new ReportCommands());
-      node.registerCommands(new ListCommands(vanishManager));
+  private void registerEvents(Object listener) {
+    if (listener instanceof Listener) {
+      getServer().getPluginManager().registerEvents((Listener) listener, this);
     }
```
[the remark this answers is not in the document]
Comment (Electroid) [reply core/src/main/java/tc/oc/pgm/PGMPlugin.java:328]: `MatchManager` does not implement `Listener`, but `MatchManagerImpl` does. It's also a clever trick around not registering `nulls (`MatchTabManager` can be null, if the tablist is disabled)

[the remark this answers is not in the document]
Comment (KingOfSquares) [reply core/src/main/java/tc/oc/pgm/PGMPlugin.java:328]: Oh yeah! That is smart :smile:!

**Pull request merged**